src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/InstructionEncoderTests.cs

```diff
@@ -475,6 +475,35 @@ public void Branch()
             }, builder.ToArray());
         }
 
+        [Fact]
+        public void Switch()
+        {
+            var builder = new BlobBuilder();
+            var controlFlowBuilder = new ControlFlowBuilder();
+            var il = new InstructionEncoder(builder, controlFlowBuilder);
+            var l = il.DefineLabel();
+            var switchEncoder = il.Switch(4);
```

Comment:
Should we be validating an exception is thrown if Switch is called again before the right number of branches have been output?

Reply:
I added validation that it throws if we call `OpCode`. `Switch` could be considered a special case of it, otherwise we would need asserts for all other shorthand methods like `Load***`, `Call` and `Branch`.